crates/starknet-os-types/src/contract_class.rs

```diff
@@ -0,0 +1,170 @@
+use std::cell::OnceCell;
+use std::rc::Rc;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::hash::GenericClassHash;
+
+pub type CairoLangCasmClass = cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+pub type BlockifierCasmClass = blockifier::execution::contract_class::ContractClassV1;
+
+#[derive(thiserror::Error, Debug)]
+pub enum ContractClassError {
+    #[error("Internal error: no type conversion is possible to generate the desired effect.")]
+    NoPossibleConversion,
+
+    #[error("Could not build Blockifier contract class")]
+    BlockifierConversionError,
+
+    #[error(transparent)]
+    SerdeError(#[from] serde_json::Error),
+}
+
+/// A generic contract class that supports conversion to/from the most commonly used
+/// contract class types in Starknet and provides utility methods.
+/// Operations are implemented as lazily as possible, i.e. we only convert
+/// between different types if strictly necessary.
+/// Fields are boxed in an RC for cheap cloning.
+#[derive(Debug, Clone)]
+pub struct GenericCasmContractClass {
```

Comment:
No because I don't want to instantiate all the types inside this in all code paths. There may be cases where we can just instantiate the `cairo-lang` variant and do everything with it, avoiding the generation of the Blockifier variant.